glossary.Rmd

```diff
@@ -0,0 +1,163 @@
+
+```{r setup, include = FALSE}
+source("setup.R")
+library("dplyr")
+```
+
+# Glossary
+
+This glossary contains all the vocabulary necessary to work with tidy evaluation and, more generally, with [expressions](#glossary-expr). The definitions in rlang are generally consistent with base R. When they differ, both definitions are presented so you can navigate between these two worlds more easily.
+
+
+## Data structures
+
+### TODO Data mask {#glossary-data-mask}
+
+### Expression {#glossary-expr}
+
+An expression is a piece of R code that describes a value:
+
+```{r}
+12
+```
+
+or a computation:
+
+```{r}
+12 / 3
+```
+
+Expressions are normally transient. They are executed ([evaluated](#glossary-eval)) when you source a file or call a function and you only see their final value or their side effects (such as console output). However in R you can suspend the normal evaluation of expressions with the [quotation](#glossary-quotation) mechanism:
```

Comment:
How about *freeze* as an analogy to quotation? This is now:

In R however, it is possible to suspend the normal evaluation of expressions with the [quotation](#glossary-quotation) mechanism. In a way, quotation causes expressions to freeze in place:

Reply:
I think @hadley uses "capture" in Advanced R. I prefer that to freeze only because freeze sounds less "portable" somehow—you're quoting something with the intent that it will be used elsewhere later (possibly). I get the freeze analogy, it's not a bad one, I just think it's somehow more prone to misinterpretation.

Reply:
When I freeze a pizza I sure intend to use it later :)

Reply:
Hahaha totally. It's just that we often use "freeze" and "frozen" in a way that means ~stuck as opposed to "saved for later."

Reply:
I think we need some kind of analogy that suggests the suspension of evaluation/computation, cf thoughts in https://github.com/tidyverse/tidyeval/issues/10#issuecomment-419851981

Reply:
I would like to explore the freeze analogy for now. I'll draft a quotation-evaluation section where we can continue the discussion. I'll present evaluation as a process of computation. There are three ways of preventing the computation of R expressions: not calling a function, not using an argument (because of lazy evaluation), and quoting. The latter is different from the first two because the argument should have been computed, but wasn't because of a user action (while in the first two cases the expressions were not computed because of a *lack* of action).